File: squares/get_squares.rb
require 'pry'

num_array = [4, 1, 16, 1, 10, 35, 22]

def get_squares(num_array)
  square_array = []
  num_array.each do |number|
    square = Math.sqrt(number)
    if square % 1 == 0
      square_array.push(number)
    end
  end
  square_array.uniq.sort
end

binding.pry

File: squares/getSquares.js
var numArray = [4, 1, 16, 1, 10, 35, 22];

var unique = numArray.filter(function(elem, index, self) {
    return index == self.indexOf(elem);
});

var getSquares = function() {
  squareArray = [];
  unique.forEach(function(number){
    if (Math.sqrt(number) % 1 == 0) {
      squareArray.push(number)
    }
  });
  return squareArray.sort(function(a, b){return a-b});
}
